server/konnectors/edf.coffee

```diff
@@ -0,0 +1,1321 @@
+            client_id: requiredFields.edeliaClientId
+            grant_type: 'edf_sso'
+            jeton_sso: data.edfToken
+            bp: entries.client[0].clientId
+            # TODO : one procedure per contract !!!
+            pdl: entries.contract[0].pdl
+        json: true
+    , (err, response, result) ->
+        if err
+            K.logger.error 'While fetching edelia token.'
+            K.logger.error err
+            return callback err
+
+        K.logger.info 'Fetched edelia token'
+        data.edeliaToken = result.access_token
+        data.contract = entries.contract[0]
+        callback()
+
+
+fetchEdeliaProfile = (requiredFields, entries, data, callback) ->
+    K.logger.info "fetchEdeliaMonthlyProfile"
+    getEdelia data.edeliaToken, '/sites/-/profiles/simple?ts=' +
+    new Date().toISOString(), (err, response, obj) ->
+        error = null
+        try
+            err = 'no import performed' if not err and not obj
+
+            if err
+                K.logger.error 'While fetchEdeliaProfile'
+                K.logger.error err
+                throw err
+
+            if obj.errorCode and obj.errorCode is "403"
+                data.noEdelia = true
+                K.logger.warn "No edelia: #{obj.errorDescription}"
+                throw null
+
+            doc =
+                pdl: data.pdl
+                beginTs: obj.beginTs
+                isProfileValidated: obj.isProfileValidated
+                housingType: obj.housingType
+                residenceType: obj.residenceType
+                occupationType: obj.occupationType
+                constructionDate: obj.constructionDate
+                isBBC: obj.isBBC
+                surface: obj.surfaceInSqMeter
+                occupantsCount: obj.noOfOccupants
+                principalHeatingSystemType: obj.principalHeatingSystemType
+                sanitoryHotWaterType: obj.sanitoryHotWaterType
+                docTypeVersion: K.docTypeVersion
+
+            entries.home.push doc
+            K.logger.info 'Fetched fetchEdeliaProfile'
+
+        catch e
+            error = e
+        finally
+            callback error
+
+
+##
+# Edelia electricite
+##
+
+fetchEdeliaMonthlyElecConsumptions = (
+requiredFields, entries, data, callback) ->
+    return callback() if data.noEdelia
+
+    K.logger.info "fetchEdeliaMonthlyElecConsumptions"
+    getEdelia data.edeliaToken, "/sites/-/monthly-elec-consumptions?" +
+    "begin-month=2012-01&" +
+    "end-month=#{moment().add(1, 'month').format('YYYY-MM')}&ended=false"
+    , (err, response, obj) ->
+
+        error = null
+        try
+            if response.statusCode is 404 or response.statusCode is 500
+                K.logger.warn 'No EdeliaMonthlyElecConsumptions'
+                data.noElec = true
+                throw null
+
+            if err
+                K.logger.error 'Wihle fetchEdeliaMonthlyElecConsumptions'
+                K.logger.error err
+                throw callback err
```

Comment:
Why throw here ? Why not just `return callback err`, or just throw err  as the callback is called in the `catch` statement